prov/util/src/util_av.c

```diff
@@ -552,7 +552,9 @@ int ofi_av_init_lightweight(struct util_domain *domain, const struct fi_av_attr
 	 */
 	av->context = context;
 	av->domain = domain;
-	ofi_genlock_init(&av->ep_list_lock, OFI_LOCK_MUTEX);
+	ofi_genlock_init(&av->ep_list_lock,
+			 domain->threading == FI_THREAD_DOMAIN ?
+			       OFI_LOCK_NOOP : OFI_LOCK_MUTEX);
```

Comment:
The original commit says it was fixing the API per Sean's concern about the ep_list_lock needing to be thread safe regardless of the threading level. I don't think reverting that fix in the av and cq is the proper answer here.

Reply:
I think it is clearer if you look at the files changed tab instead of commit by commit.